pkg/identity/cache/cache.go

```diff
@@ -184,6 +195,13 @@ func (w *identityWatcher) watch(events allocator.AllocatorEventRecvChan) {
 			}
 			// Issue collected updates
 			w.owner.UpdateIdentities(added, deleted) // disjoint sets
+
+			if synced && !closed {
+				w.logger.Info("All initial identities added to SelectorCache.")
+				close(w.watchDone)
```

Comment:
Given that this is unblocked once local synchronization completed, I assume we may have the very same race condition for remote identities when clustermesh is enabled. Will that be addressed in a follow-up PR?

Reply:
That gave me an idea for how to fix this: the events themselves can include a done channel that is closed when it's processed. This then delays the Allocator's status.

Let me try this out. Solves the clustermesh problem neatly.